### ISSUE TITLE
Search in linking starts displaying results after entering 3 characters for text based search

### ISSUE THREAD

**Issue by naina-verma** [almighty/almighty-ui#632]
Search in linking starts displaying results after entering 3 characters  
But the id based search starts with 1st digit ,
Video : https://drive.google.com/a/redhat.com/file/d/0B1qUq_laG6CObnVCckFaWUxpUlk/view?usp=sharing
Is this by design or unintentional behavior !

**Comment by Mgranfie** [comment]
@naina-verma @nimishamukherjee I have posted a solution for the lag when waiting on a 3 character entry in the type ahead. Please review and provided feedback.

https://redhat.invisionapp.com/share/M28SSTLRJ#/215405218_DesktopAddLink_04_DD_TypeAhead

**Comment by Mgranfie** [comment]
@sanbornsen @naina-verma @nimishamukherjee I have also included here the general drop down behaviors, as I was not able to find another story that related to these. Please review and questions and feedback welcome.

https://redhat.invisionapp.com/share/M28SSTLRJ#/215402961_DesktopAddLink_04_DD_Behaviors

**Comment by naina-verma** [comment]
Hi @nimishamukherjee  @Mgranfie ,
I looked into [invision](https://redhat.invisionapp.com/share/M28SSTLRJ#/215405218_DesktopAddLink_04_DD_TypeAhead) , i think we should support some thing like this search  : 
https://fscl01.fonpit.de/userfiles/7176834/image/whatsapp-search-w782.jpg
This is from very know Application  :smile:  please give your thoughts !

**Comment by Mgranfie** [comment]
Hi Nania thanks for sending.

You are referring to highlighting the text in the results with the search
criteria?Yes this is a pattern we use in the type ahead for Assign user and
in the meeting someone mentioned (Sudipta?) that we would incorporate this
here too.

It would be great to have it here too!
